### PR TITLE
[#7] Setup development environment.

### LINKED ISSUE #7: Setup development environment

### PR DESCRIPTION
**Pull request by qthiendev**
This pull request sets up the development environment for both the **Laravel 11.31** backend and **Vite 6.0.1** with React frontend. The focus is on configuring the folder structure, environment variables, and basic setup for each component. No connections between the frontend and backend are established yet in this PR.

Closes #7

## **Changes Included**

1. **Frontend (Vite 6.0.1 Setup)**:

   - Added **Vite 6.0.1** as the build tool for the frontend.
   - Configured `vite.config.js` to handle the React app.
   - Created the `client` directory for the React project.

3. **Backend (Laravel 11.31 Setup)**:

   - Created the `server` directory for the Laravel backend.
   - Set up a basic **Laravel 11.31** installation.
   - Added initial `.env` configuration for local environment setup.
   - Set up database configuration and environment variables for Laravel.

4. **Folder Structure**:

   - Organized the project into `client` (for the React frontend) and `server` (for the Laravel backend).
   - Ensured that the `client` and `server` directories are separate for clear project structure.
   - Added basic files such as `package.json` for React and `composer.json` for Laravel.

5. **Environment Setup**:

   - Configured `.env` files for both React and Laravel.
   - Set up basic local environment variables for development purposes (e.g., `DB_HOST`, `APP_ENV`, etc.).

## **Why This PR Is Needed**

This PR establishes the foundational setup for the development environment, ensuring that:
- Developers can start working with both the Laravel backend and React frontend.
- The project structure is organized with separate directories for frontend and backend.
- Basic configurations for local development are in place.

## **How To Test**

1. **Setup**:
   - Open `cmd`
   - Clone this branch and navigate to repo folder.
   - Run command:
      ```bash
      php composer.phar install --working-dir=./server
      cd client
      npm install
      cd ..
      ```
   - Split `cmd` or open new `cmd`:
   - One cmd run:
      ```bash
      cd server
      php artisan serve
      ```
   - Other cmd run:
      ```bash
      cd client
      npm run dev
      ```

### Setup `server`:

image

image

### Setup `client`:

image

### Run `server`:

image

### Run `client`:

image

## **Additional Notes**
- Please ensure you have PHP 8.x and Node.js installed on your local machine.
- The `.env` file should be configured with the correct database and API settings for local development.

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,21 @@
+MIT License
```
Comment (LeVanCan) [inline LICENSE:1]: @hungleba-dtvn 
Nhờ a xem giúp e cái LICENSE này có vấn đề gì ở đây k ạ?

File: client/index.html [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+  <title>E-Learning</title>
+</head>
+
+<body>
+  <div id="root"></div>
+  <script type="module" src="/src/main.jsx"></script>
+</body>
+
+</html>
```
Comment (LeVanCan) [inline client/index.html:16]: Please add EOL at the end of file, other file is same.

File: LICENSE (hunk 1 above)
Comment (qthiendev) [reply LICENSE:1]: @hungleba-dtvn dạ cái này nó tạo default ạ, nếu không cần thiết để em xóa ở commit update sau ạ

**Review by LeVanCan — APPROVED** [review]
LGTM

**Pull request merged**